lib/Runtime/Base/ThreadContext.h

```diff
@@ -650,6 +652,22 @@ class ThreadContext sealed :
     uint functionCount;
     uint sourceInfoCount;
 
+    enum RedeferralState
+    {
+        InitialRedeferralState,
+        StartupRedeferralState,
+        MainRedeferralState
+    };
+    RedeferralState redeferralState;
+    uint gcSinceLastRedeferral;
+    uint gcSinceCallCountsCollected;
+
+    static const uint InitialRedeferralDelay = 5;
```

Comment:
>InitialRedeferralDelay [](start = 22, length = 22)

should we have these as configurable values so we can tune redeferral in the future if need be?